drift/core/beamtransfer.py

```diff
@@ -1536,116 +1606,119 @@ def _generate_svdfiles(self, regen=False):
-                bft = bf[:, 0, :]
+                    # Read the positive and negative m beams, and combine into one.
+                    with h5py.File(self._mfile(mi), "r") as fm:
+                        bf = fm["beam_m"][fi][:].reshape(
+                            self.ntel,
+                            self.telescope.num_pol_sky,
+                            self.telescope.lmax + 1,
+                        )
 
-                # Perform the SVD to find the left evecs
-                u, sig, v = svd_gen(bft, full_matrices=False)
-                u = u.T.conj()  # We only need u^H so just keep that.
+                    noisew = self.telescope.noisepower(
+                        np.arange(self.telescope.npairs), fi
+                    ).flatten() ** (-0.5)
+                    noisew = np.concatenate([noisew, noisew])
+                    bf = bf * noisew[:, np.newaxis, np.newaxis]
 
-                # Save out the evecs (for transforming from the telescope frame into the SVD basis)
-                dset_ut[fi] = u * noisew[np.newaxis, :]
+                    # Get the T-mode only beam matrix
+                    bft = bf[:, 0, :]
 
-                # Save out the modified beam matrix (for mapping from the sky into the SVD basis)
-                bsvd = np.dot(u, bf.reshape(self.ntel, -1))
-                dset_bsvd[fi] = bsvd.reshape(
-                    self.svd_len, self.telescope.num_pol_sky, self.telescope.lmax + 1
-                )
+                    # Perform the SVD to find the left evecs
+                    u, sig, v = svd_gen(bft, full_matrices=False)
+                    u = u.T.conj()  # We only need u^H so just keep that.
 
-                # Find the pseudo-inverse of the beam matrix and save to disk.
-                dset_ibsvd[fi] = la.pinv(bsvd).reshape(
-                    self.telescope.num_pol_sky, self.telescope.lmax + 1, self.svd_len
-                )
+                    # Save out the evecs (for transforming from the telescope frame into the SVD basis)
+                    dset_ut[fi] = u * noisew[np.newaxis, :]
 
-                # Save out the singular values for each block
-                dset_sig[fi] = sig
+                    # Save out the modified beam matrix (for mapping from the sky into the SVD basis)
+                    bsvd = np.dot(u, bf.reshape(self.ntel, -1))
+                    dset_bsvd[fi] = bsvd.reshape(
+                        self.svd_len,
+                        self.telescope.num_pol_sky,
+                        self.telescope.lmax + 1,
+                    )
 
-            # Write a few useful attributes.
-            fs.attrs["baselines"] = self.telescope.baselines
-            fs.attrs["m"] = mi
-            fs.attrs["frequencies"] = self.telescope.frequencies
-            fs.attrs["cylobj"] = self._telescope_pickle
+                    # Find the pseudo-inverse of the beam matrix and save to disk.
+                    dset_ibsvd[fi] = la.pinv(bsvd).reshape(
```

Comment:
do we know is this pseudo inverse reliably? I never use this class, otherwise we might also have a try error block here with pinv2.

Reply:
Indeed, I also never use this `BeamTransferTempSVD` class, so I haven't done any testing of it - the only change I made to its code is a minor re-arrangement, rather than the more extensive changes in `BeamTransfer`.